### ISSUE TITLE
onWebsocketHandshakeReceivedAsServer throwing InvalidDataException has no effect

### ISSUE THREAD

**Issue by devgianlu** [TooTallNate/Java-WebSocket#530]
I've implemented this piece of code on my WebSocketServer:
```
public ServerHandshakeBuilder onWebsocketHandshakeReceivedAsServer(WebSocket conn, Draft draft, ClientHandshake request) throws InvalidDataException {
        ServerHandshakeBuilder builder = super.onWebsocketHandshakeReceivedAsServer(conn, draft, request);
        if (!shouldAccept(request)) {
            throw new InvalidDataException(CloseFrame.POLICY_VALIDATION, "Not accepted!");
        }

        return builder;
    }
```

When an invalid handshake is sent by the client, nothing happens. The client doesn't close the connection nor connect.

**Comment by marci4** [comment]
Hello @devgianlu,

thx for your bug report!

Looks like an internal error which is caused by WebSocketImpl between 271 to 275. I will look into this!

Greetings
marci4

**Comment by devgianlu** [comment]
I've done some more debugging:

- throwing a RuntimeException doesn't work either
- I can close the connection with WebSocket#close(int, String) but the close code and the message are overwritten 
- If I call WebSocket#closeConnection(int, String), the connection is closed but with close code -1.

**Comment by marci4** [comment]
Thx again for the time you invest into this issue!

Hope I get the time to check this out over the weekend ;)

Could also be just the flushAndClose (since there are no bytes to write maybe this is causing problems).

Greetings
marci4

**Comment by marci4** [comment]
Hello @devgianlu,
I fixed your issue with pull request #532.

Closing this issue!
Greetings
marci4
